### ISSUE TITLE
Referencing a compile time variable in a lexicographically independent manner

### ISSUE THREAD

**Issue by ephja** [nim-lang/Nim#6050]
#6041 allows a compile time variable to be used at run time without the use of constants, but the value that the compile time symbol evaluates to depends on where it is being referenced lexically.

However, it might be useful, in some cases, to refer to the value that this compile time symbol holds just at the very end of the compile time context, such that the test in the aforementioned PR would output this instead: "3\n3\n3\n3"; possibly after annotating each use of the compile time variable with some pragma that enables this proposed behavior.

**Comment by bkerin** [comment]
Yeah, I just started wanting exactly this.  It make sense intuitively ("give me the value that fooVar ended up with at the end of compilation") and is likely to be highly useful when you have some run-time work to
do to correspond to work done at compile time.
